client/tests/integration_test.rs

```diff
@@ -0,0 +1,160 @@
+    let builder = Server::try_bind(&([127, 0, 0, 1], port).into()).unwrap();
+    let make_svc =
+        hyper::service::make_service_fn(move |_| future::ok::<_, Infallible>(service.clone()));
+    let (graceful_shutdown, shutdown_receiver) = oneshot::channel::<()>();
+    let server = builder.serve(make_svc);
+    let server_with_shutdown = server.with_graceful_shutdown(async {
+        shutdown_receiver.await.ok();
+    });
+    let server_joiner = tokio::spawn(server_with_shutdown);
+    MockServerHandle {
+        graceful_shutdown,
+        server_joiner,
+    }
+}
+
+macro_rules! assert_valid_response {
+    ($response: expr) => {
+        assert_eq!(
+            $response
+                .expect("parsed request values")
+                .get_result()
+                .map(|val| serde_json::from_value::<()>(val.clone()).unwrap()),
+            Some(())
+        );
+    };
+}
+
+#[tokio::test(threaded_scheduler)] // needed to spawn the mock server in the background
+async fn client_should_send_get_state_root_hash() {
+    let node_port = 8999;
+    let handle =
+        spawn_mock_rpc_server::<GetStateRootHashParams>(node_port, GetStateRootHash::METHOD);
+
+    let response = casper_client::get_state_root_hash(
+        "1",
+        &format!("http://localhost:{}", node_port),
+        false,
+        "09dcee4b212cfd53642ab323fbef07dafafc6f945a80a00147f62910a915c4e6",
+    );
+    assert_valid_response!(response);
+
+    handle.kill().await;
+}
+
+#[tokio::test(threaded_scheduler)] // needed to spawn the mock server in the background
+async fn client_should_send_get_deploy() {
+    let node_port = 9001;
+    let handle = spawn_mock_rpc_server::<GetDeployParams>(node_port, GetDeploy::METHOD);
+
+    let response = casper_client::get_deploy(
+        "1",
+        &format!("http://localhost:{}", node_port),
+        false,
+        "09dcee4b212cfd53642ab323fbef07dafafc6f945a80a00147f62910a915c4e6",
+    );
+    assert_valid_response!(response);
+
+    handle.kill().await;
+}
+
+#[tokio::test(threaded_scheduler)]
+async fn client_should_send_put_deploy() {
+    let node_port = 9002;
+    let handle = spawn_mock_rpc_server::<PutDeployParams>(node_port, PutDeploy::METHOD);
+
+    let response = casper_client::put_deploy(
+        "1",
+        &format!("http://localhost:{}", node_port),
+        false,
+        DeployStrParams {
+            secret_key: "../resources/local/secret_keys/node-1.pem",
+            ttl: "10s",
+            chain_name: "casper-test-chain-name-1",
+            gas_price: "1",
+            ..Default::default()
+        },
+        SessionStrParams::with_path(
+            "../target/wasm32-unknown-unknown/release/standard_payment.wasm",
+            vec!["name_01:bool='false'", "name_02:i32='42'"],
+            "",
+        ),
+        PaymentStrParams::with_amount("100"),
+    );
+    assert_valid_response!(response);
+    handle.kill().await;
+}
+
+#[tokio::test(threaded_scheduler)]
+#[should_panic(
```

Comment:
Let's not panic when we don't have to, since it stops us continuing within the same test, it's less canonical, and it pollutes the output of successful test runs.  The test can check the `Error` returned without panicking.